include/NAS2D/DataUtils.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "NAS2D/StringUtils.h"
+
+#include <string>
+
+namespace NAS2D
+{
+	namespace Xml
+	{
+		class XmlElement;
+
+		void validateXMLElement(
+			const XmlElement& element,
+			const std::string& name,
+			const std::string& requiredChildElements,
+			const std::string& requiredAttributes,
+			const std::string& optionalChildElements = std::string{},
+			const std::string& optionalAttributes = std::string{});
```

Comment:
I suppose a matter of style, though I might suggest spilling the closing parenthesis onto it's own line:
```cpp
void validateXMLElement(
	const XmlElement& element,
	const std::string& name,
	const std::string& requiredChildElements,
	const std::string& requiredAttributes,
	const std::string& optionalChildElements = std::string{},
	const std::string& optionalAttributes = std::string{}
);
```
That way the indentation is a little more consistent with the parsed structure.

Reply:
I blame Visual Studio. I just spent more than a few seconds trying to get it to not do that in a different file.